### PR TITLE
Feature/redux

### PR DESCRIPTION
**Pull request by nottherealalanturing**
## Describe your changes
Setting up project repo with CRA
- Added the Redux and reduxtoolkit library
- Added reducers and actions for adding and removing books
- Created the global store

### What kind of change does this PR introduce?
New Feature.

### In which branch do you want the changes to merge?

dev.

### Documentation updated ?

Yes

### Checklist

- [ ] No ES Lint Error
- [ ] No Style Lint Error
- [ ] Unit test case update

###Screenshot

Screen Shot 2022-06-04 at 14 25 38

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+export const addBook = (book) => ({
+  type: 'books/addBook',
+  payload: book,
+});
+
+export const deleteBook = (book) => ({
+  type: 'books/remove',
+  payload: book,
+});
```
Comment (keroloussamy) [inline src/redux/books/books.js]: - [x] Kindly to follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux), you can add the action type like that: `const ADD_BOOK = bookstore/books/ADD_BOOK` and `const REMOVE_BOOK = bookstore/books/REMOVE_BOOK` and after that use them in your file.

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const initialState = [];
+
+export const checkStatus = () => ({
+  type: 'categories/checkStatus',
+});
```
Comment (keroloussamy) [inline src/redux/categories/categories.js]: - [x] Kindly to follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux), you can add the action type like that: `const CHECK_STATUS = bookstore/categories/CHECK_STATUS` and after that use them in your file.

**Review by keroloussamy — CHANGES_REQUESTED** [review]
Hi @nottherealalanturing, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️

## Required Changes ♻️
_Check the comments under the review._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by m05e5 — APPROVED** [review]
# Hi @nottherealalanturing  , 

### Status: Approved ✔️
goodjob


:+1: . Well Done.

Your project is complete! 
Congratulations! 🎉 


**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
-----

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_m05e5_**

------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**